test/test_proxymanager.py

```diff
@@ -57,3 +60,10 @@ def test_proxy_tunnel(self):
         with ProxyManager("https://proxy:8080") as p:
             assert p._proxy_requires_url_absolute_form(http_url)
             assert p._proxy_requires_url_absolute_form(https_url)
+
+    def test_proxy_connect_retry(self):
```

Comment:
Can we get a test that shows that a ProxyError caused a retry?

Reply:
Added :)